src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java

```diff
@@ -0,0 +1,382 @@
+package com.hms.flexyosisoftconnector;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+
+import com.hms.flexyosisoftconnector.PayloadBuilder;
+import com.hms_networks.americas.sc.datapoint.DataPoint;
+import com.hms_networks.americas.sc.json.JSONException;
+import com.hms_networks.americas.sc.logging.Logger;
+import com.hms_networks.americas.sc.string.PreAllocatedStringBuilder;
+import com.hms_networks.americas.sc.taginfo.TagConstants;
+import com.hms_networks.americas.sc.taginfo.TagInfoManager;
+
+/**
+ * This class will hold a payload string and a status
+ *
+ * <p>The payload string will be appended to until the status has been set to done
+ *
+ * @author HMS Networks Inc. Solution Center
+ */
+public class OsisoftJsonPayload {
+
+  /**
+   * Current payload status
+   */
+  int status;
+
+  /**
+   * String builder for payload
+   */
+  final PreAllocatedStringBuilder payload;
+
+  /**
+   * Constant representing payload not started
+   */
+  static final int PAYLOAD_NOT_STARTED = 0;
+
+  /**
+   * Constant representing payload header complete
+   */
+  static final int PAYLOAD_HEADER_COMPLETE = 1;
+
+  /**
+   * Constant representing payload values in progress
+   */
+  static final int PAYLOAD_VALUES_IN_PROGRESS = 2;
+
+  /**
+   * Constant representing payload values complete
+   */
+  static final int PAYLOAD_VALUES_COMPLETE = 3;
+
+  /**
+   * Constant representing payload completion
+   */
+  static final int PAYLOAD_COMPLETE = 4;
+
+  /**
+   * Constant representing how many datapoints can be added to a payload
+   */
+  static final int MAX_DATA_POINTS = 200;
+
+  /**
+   * The unfinished payload will have an array of strings. Each tag has it's own array that gets
+   * added to when the tag is encountered.
+   */
+  TagPayload[] tagPayloadArr;
+
+  /**
+   * Array of boolean flags for tags encountered (by ID)
+   */
+  boolean[] tagNamesEncountered;
+
+  /**
+   * Count of tags encountered
+   */
+  int dataPointsEncounteredCount = 0;
+
+  /**
+   * Offset for indexing tags in {@link #tagNamesEncountered}
+   *
+   * <p>This offset will be equal to the lowest valued tag ID found.
+   */
+  int tagIndexOffset;
+
+  /**
+   * Size of encountered tags array
+   */
+  int tagsEncounteredArraySize;
+
+  /**
+   * Timestamp for the start of payload
+   *
+   * <p>This timestamp means that payload datapoint can come from any time from the timestamp
+   * listed
+   * up to an additional {@link #PAY_LOAD_TIME_PERIOD_SECONDS} seconds. Payloads are from a time
+   * period, not a single time stamp.
+   */
+  long payloadStartTimestamp = 0;
+
+  /**
+   * This is the number of seconds to wait before cutting off the payload and completeing it.
+   */
+  static final long PAY_LOAD_TIME_PERIOD_SECONDS = 10;
+
+  /**
+   * represents either OMF or the lagacy data format.
+   */
+  int communicationType;
+
+  String comsErrMsg = "Unknown communication type given. Unable to build payload.";
```

Comment:
missing javadoc

Reply:
I'm adding in a couple of java docs that i missed alongside this

Reply:
I was just starting a review, should I hold off till the comments are updated?

Reply:
sure, I'll push it again in maybe 10 minutes

Reply:
I may have got them all, it's good to go @TomKimsey